mir_dataset_loaders/salami.py

```diff
@@ -0,0 +1,194 @@
+    return missing_files, invalid_checksums
+
+
+def track_ids():
+    return list(SALAMI_INDEX.keys())
+
+
+def load(data_home=None):
+    validate(data_home)  # TODO: when missing files should avoid loading them
+    salami_data = {}
+    for key in track_ids():
+        salami_data[key] = load_track(key, data_home=data_home)
+    return salami_data
+
+
+def load_track(track_id, data_home=None):
+    if track_id not in SALAMI_INDEX.keys():
+        raise ValueError(
+            "{} is not a valid track ID in Salami".format(track_id))
+
+    if SALAMI_METADATA is None or SALAMI_METADATA['data_home'] != data_home:
+        _reload_metadata(data_home)
+        if SALAMI_METADATA is None:
+            raise EnvironmentError("Could not find Salami metadata file")
+
+    if track_id in SALAMI_METADATA.keys():
+        track_metadata = SALAMI_METADATA[track_id]
+    else:
+        # annotations with missing metadata
+        track_metadata = {'source': None, 'annotator_1_id': None, 'annotator_2_id': None, 'duration_sec': None,
+                          'title': None, 'artist': None, 'annotator_1_time': None, 'annotator_2_time': None,
+                          'class': None, 'genre': None}
+
+    annotations_dir = os.path.join(data_home, 'salami-data-public-master','annotations')
+    annotators = [any(SALAMI_INDEX[track_id]['annotator_1_uppercase']),
+                  any(SALAMI_INDEX[track_id]['annotator_2_uppercase'])]
+    sections_data = _load_sections(get_local_path(annotations_dir, track_id), annotators)
+
+    return SalamiTrack(
+        track_id,
+        sections_data[0],
+        sections_data[1],
+        sections_data[2],
+        sections_data[3],
+        track_metadata['source'],
+        track_metadata['annotator_1_id'],
+        track_metadata['annotator_2_id'],
+        track_metadata['duration_sec'],
+        track_metadata['title'],
+        track_metadata['artist'],
+        track_metadata['annotator_1_time'],
+        track_metadata['annotator_2_time'],
+        track_metadata['class'],
+        track_metadata['genre']
+    )
+
+
+def _load_sections(sections_path, annotators):
+    sections_data = []
+    for a in range(len(annotators)):
+        times, secs = [], []
+        for f in ['uppercase.txt', 'lowercase.txt']:
+            if annotators[a]:
+                file_path = os.path.join(sections_path, 'parsed',
+                                         'textfile{}_{}'.format(str(a + 1), f))
+                if os.path.exists(file_path):
+
+                    with open(file_path, 'r') as fhandle:
+                            reader = csv.reader(fhandle, delimiter='\t')
+                            for line in reader:
+                                times.append(float(line[0]))
+                                secs.append(line[1])
+
+                    sections_data.append(SectionsData(np.array(times)[:-1],
+                                                      np.array(times)[1:], np.array(secs)))
```

Comment:
The difference was in the `End` label which is an instant, included begin and end time the same for that label

Reply:
Should the return then instead be `np.array(secs[:-1])`?

Reply:
I'm keeping the last section label, just assigning it the same start and end (so all arrays are of same length)

Reply:
I think I'm still missing something - I don't see from your code how the arrays `np.arrays(times)[:-1]` and `np.array(secs)` can be the same length, when the lists `times` and `secs` are the same lengths.

Reply:
At the end removed the last label (which had no duration and corresponded to the label 'End'). Also remove sections with duration == 0, otherwise it breaks `mir_eval`.